### ISSUE TITLE
disabled prop does not work for Button and Radio despite the type allowing it 

### ISSUE THREAD

**Issue by grzmot22** [chakra-ui/chakra-ui#7269]
### Description

The type for Button and Radio allows both the disabled and isDisabled props, but only isDisabled works. disabled will be silently ignored with no warning whatsoever. 

Please fix this by either changing the type so disabled is no longer a prop of  Button, Radio or making it so that disabled works.

### Link to Reproduction

https://codesandbox.io/s/intelligent-jang-wu06rc

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Chakra UI Version

2.4.9

### Operating System

- [x] macOS
- [x] Windows
- [x] Linux

**Comment by GKosheev** [comment]
This change occurred 6 days ago (`"refactor: button disabled prop"`): 

```diff
<chakra.button
- disabled={isDisabled || isLoading}
  ref={useMergeRefs(ref, _ref)}
  as={as}
  type={type ?? defaultType}
  data-active={dataAttr(isActive)}
  data-loading={dataAttr(isLoading)}
  __css={buttonStyles}
  className={cx("chakra-button", className)}
  {...rest}
+  disabled={isDisabled || isLoading}
>
```

I can't make any comments on this change, but I know that before this change, you could use both `disabled` and `isDisabled`, which would lead to uncertain behavior. For example, if you use `isDisabled` or `isLoading`, you can expect the button to be disabled, but you can override that behavior by setting `disabled` to `false`: 

```tsx
<Button isLoading disabled={false}>
  Button
</Button>
```

**Comment by ZiggStardust** [comment]
i have the same issue

**Comment by samuliasmala** [comment]
I'd be happy to create a PR but first I'd need a comment from maintainers whether I should update the functionality (i.e. make `disabled` prop to work again) or the typing.

**Comment by grzmot22** [comment]
I noticed was disabled prop was changed in many components :/

**Comment by aoloo** [comment]
I noticed this issue as well. It seems the `disabled` attribute is not being applied to the button

**Comment by Trisco** [comment]
+1

**Comment by RobinClowers** [comment]
Yeah, this is a breaking change that was called a "refactor" https://github.com/chakra-ui/chakra-ui/pull/7233

**Comment by RobinClowers** [comment]
This is the actual commit: https://github.com/chakra-ui/chakra-ui/commit/0f376650efce9eb2d2cba3484ffe4e3a73364504

**Comment by RobinClowers** [comment]
Here is the original issue that prompted the change: https://github.com/chakra-ui/chakra-ui/issues/7243

**Comment by grzmot22** [comment]
@RobinClowers Great you found the original issue, but I think it needs to be better handled not to confuse people.  Shouldn't be two props that are doing the same thing. I'm curious why were two and `disabled` is more common than `isDisabled`. I could do PR, but I don't know the idea behind having another prop, so it is better to leave it for the creators of Chakra ui.

**Comment by jrolfs** [comment]
The `disabled` prop was passed directly to the `<button />` element without updating the styles to the disabled state specified in the theme. We're using this to disable the button while in a loading state that's not covered by `isLoading`, so that's a use case for having them separate. I'm not saying it's a case that needs to be supported, but for us this was a breaking change.

**Comment by samuliasmala** [comment]
According to [this comment](https://github.com/chakra-ui/chakra-ui/issues/7243#issuecomment-1383939865) it's the typings which should be updated, but @segunadebayo can you confirm it?

**Comment by jrolfs** [comment]
> but for us, this was a breaking change.

Is this not a concern? I get that the surface area for breaking changes in a component library like this is massive and difficult to pin down, but since this was included in the types, I feel it's reasonable to consider this a breaking change.

**Comment by anilanar** [comment]
Not sure what % of users will be hit by this, but my feeling is that this issue is urgent and will blow up further when more users try to upgrade and notice (or not notice) that `disabled` doesn't work anymore. @anubra266

**Comment by sleekLancelot** [comment]
Presently, the only ways to disable a button is with an "isLoading" or "isDisabled" prop.
See Segun's comment here: https://github.com/chakra-ui/chakra-ui/issues/7243#issuecomment-1383939865

**Comment by ckpearson** [comment]
Just got caught out by this. The type supporting the `disabled` prop as well as `isDisabled` is a little confusing. Might be a good idea to signpost this in the docs.

**Comment by JulioC** [comment]
@segunadebayo I believe this is not only related to TypeScript, as it's a breaking change from previous behavior.

**Comment by grzmot22** [comment]
@segunadebayo   I agree with @JulioC, even though I would like to make PR, I don't know what was intended to be not using the typical flag `disabled` and having it both. What was origin of using `isDisabled` in the first place?

**Comment by JulioC** [comment]
For reference, this is the commit which changed this behavior: https://github.com/chakra-ui/chakra-ui/commit/0f376650efce9eb2d2cba3484ffe4e3a73364504

The original issue motivating the change: https://github.com/chakra-ui/chakra-ui/issues/7243

---

I believe either solutions are acceptable (I prefer the first one):

- Change the affected line in button.tsx to be `disabled={props.disabled || isDisabled || isLoading}` (this revert the original behavior and fixes the original issue)
- Omit 'disabled' from the type for ButtonProps (this would be a breaking change)

**Comment by jhurs16** [comment]
Hi guys same issue here but I fixed it on this solution in stackoverflow. Please read on it.
https://stackoverflow.com/questions/75167380/how-to-disable-button-if-form-string-input-is-empty-react

**Comment by peppemig** [comment]
Yeah 'disabled' is not working for me right now, only 'isDisabled' prop is working

**Comment by ClaudioBzrr** [comment]
i had the same issue with "Disabled" prop, but with "isDisabled" prop worked fine

**Comment by trey-wallis** [comment]
I think it makes sense both ways. Since `isDisabled` includes the `is` prefix, you know for sure that the property accepts a boolean value. `disable` on the other-hand is more ambiguous, however it comes from HTM syntax where in order to disable a button you would use just "disable". e.g. `<button disable/>`. Most people are familiar with the HTML syntax so it wasn't very confusing.

**Comment by barrymichaeldoyle** [comment]
Is there any plan to fix this or at least get rid of `disabled` being a valid TS option? The state that it is in now is not great.

**Comment by SanariSan** [comment]
Got confused about why `disabled` not working, [button docs](https://chakra-ui.com/docs/components/button/usage) page says nothing about `isDisabled` prop existence, but mentions `disabled` keyword for custom `Box` button. Thanks to this issue I now see the solution, but that's something to be fixed or mentioned in the docs at least. 

p.s.
`@chakra-ui/react@^2.5.5`
`@chakra-ui/theme@^3.0.1`

**Comment by Laki997** [comment]
For me it still does not work even with isDisabled=true

**Comment by lakardion** [comment]
I updated chakra recently to latest and all my integration tests that were expecting disabled buttons started failing :|. Found out I had to use `isDisabled` instead?
Luckily we had a wrapper on every chakra button so I can do a hack there so `disabled` prop gets passed to isDisabled instead

**Comment by gustavolzangelo** [comment]
It's just working with isDisabled here.

**Comment by BANOnotIT** [comment]
Why changing public API is in fix?.. This is a BREAKING change btw

**Comment by servettonga** [comment]
`isDisabled` works but it's still not mentioned on the [documentation](https://chakra-ui.com/docs/components/button)

**Comment by anilanar** [comment]
@servettonga Look at [Props](https://chakra-ui.com/docs/components/button/props).

**Comment by servettonga** [comment]
> @servettonga Look at [Props](https://chakra-ui.com/docs/components/button/props).

Thanks, I didn't notice that

**Comment by tolkee** [comment]
It means that Chakra desactivated `disabled` props ?

That's kind of dangerous as all my company code was using `disabled` on their buttons, but in one day, button were not disabled anymore. (solution was to migrate to `isDisabled`, but still)

**Comment by BANOnotIT** [comment]
@Philzen I made a PR !7907 to solve disabled prop problem in Button

**Comment by samiulalimseam** [comment]
thanks

**Comment by Philzen** [comment]
> If it isn't going to be reverted, let's do what @grzmot22 's issue suggests and just remove disabled from the typescript type.

If that is done, it will manifest the current situation – which means the workaround for is no longer possible #7965 (we need to ensure `isLoading={true}` does not make the button lose focus. The "fix" for #7243 was well intended, but broke a lot of things, which is a violation of semver and hence *must* be rolled back IMHO.

**Comment by sbeben** [comment]
I'm in love with chakra but this button styling mess and now this makes it so painful to use

**Comment by Philzen** [comment]
@grzmot22 #8462 has been merged, `disabled` prop on button works again, you can test that out in the LiveCoding-view here:  https://chakra-ui-website-ptekgabqh-chakra-ui.vercel.app/docs/components/button/usage

Kindly also check if the prop works as desired on Radio as well – in my tests it seems to work again (at least visually).

If satisfied, kindly close this issue, the fix should be shipping with Chakra 2.8.3.

**Comment by shawnwz** [comment]
I have the same problem with version 2.8.2

**Comment by Philzen** [comment]
> I have the same problem with version 2.8.2

@shawnwz  v2.8.2 is from November 2023. The fix was only merged three weeks ago.

@segunadebayo is there an outlook for a patch release for v2? There were a handful of fixes merged into v2 since November and it would be great to have them available.

**Comment by vantage-ola** [comment]
> I have the same problem with version 2.8.2

i am on the same version, disabled didnt work but  isDisabled worked for me in button.

**Comment by ippeiukai** [comment]
> If satisfied, kindly close this issue, the fix should be shipping with Chakra 2.8.3.

Can we reopen this issue unless 2.8.3 is on the way?
